### DISCUSSION TITLE
Nooie PU13 Tuya Smart Plug Regularly Turns Off

### DISCUSSION THREAD

**Discussion by sshaikh** [make-all/tuya-local#2326] [General]
(Crossposted to HA forums)

I have 4 RTL8710 based Tuya Smart Plugs made by Nooie. I have successfully integrated them with Tuya Local and everything largely works okay.

However they all seem to turn off every 49 days. I’ve logged things enough to be sure about the time frame and can almost predict the event to the hour. Each plug seems to have its own timer based on when it was last turned off (so its not like they turn off at the same time. When they get the event to turn off, they will then stay off or switch back on depending on the “state after power loss” setting.

I don’t think this is HA or Tuya Local as HA regularly gets rebooted. I do have a developer account think in order to make TL work. I cannot tell if the event is locally generated (ie in the plug) or sent remotely somehow (my next step is to cut them off from the internet.

Before I go mad, has anyone else experienced this?

**Comment by make-all** [comment]
49 days is coincidentally the maximum uptime of Windows 95. IIRC it was due to a 32-bit millisecond timer wrapping, so something similar may be happening inside your Tuya device.
